aws/components/ddb/setup.ftl

```diff
@@ -0,0 +1,886 @@
+                                }
+                            }
+                        componentType=DIRECTORY_COMPONENT_TYPE
+                    /]
+
+                    [#local secretId = resources["rootCredentials"]["secret"].Id ]
+
+                    [#local ddsUsername = solution["rootCredential:SecretStore"].Username ]
+                    [#local ddsUsernameRef = getSecretManagerSecretRef(secretId, solution["rootCredential:SecretStore"].UsernameAttribute)]
+                    [#local ddsPassword = ""]
+                    [#local ddsPasswordRef = getSecretManagerSecretRef(secretId, solution["rootCredential:SecretStore"].PasswordAttribute)]
+                    [#break]
+
+                [#case SECRETSTORE_SECRET_COMPONENT_TYPE]
+                    [#local secretId = secretLink.State.Resources["secret"].Id]
+
+                    [#local ddsUsername = solution["rootCredential:SecretStore"].Username]
+                    [#local ddsUsernameRef = getSecretManagerSecretRef(secretId, solution["rootCredential:SecretStore"].UsernameAttribute)]
+                    [#local ddsPassword = ""]
+                    [#local ddsPasswordRef = getSecretManagerSecretRef(secretId, solution["rootCredential:SecretStore"].PasswordAttribute)]
+
+                    [#break]
+
+                [#default]
+                    [@fatal
+                        message="Invalid secretLink for db credentials"
+                        context={
+                            "Id" : core.RawId,
+                            "SecretLink" : solution["rootCredential:SecretStore"].Link,
+                            "SecretLinkComponentType" : (secretLink.Core.Type)!""
+                        }
+                    /]
+
+            [/#switch]
+            [#break]
+    [/#switch]
+
+    [#local hibernate = solution.Hibernate.Enabled && isOccurrenceDeployed(occurrence)]
+
+    [#local hibernateStartUpMode = solution.Hibernate.StartUpMode ]
+
+    [#local ddsRestoreSnapshot = getExistingReference(formatDependentRDSSnapshotId(ddsId), NAME_ATTRIBUTE_TYPE)]
+    [#local ddsManualSnapshot = getExistingReference(formatDependentRDSManualSnapshotId(ddsId), NAME_ATTRIBUTE_TYPE)]
+    [#local ddsLastSnapshot = getExistingReference(ddsId, LASTRESTORE_ATTRIBUTE_TYPE )]
+
+    [#local backupTags = [] ]
+    [#local links = getLinkTargets(occurrence, {}, false) ]
+    [#list links as linkId,linkTarget]
+
+        [#local linkTargetCore = linkTarget.Core ]
+        [#local linkTargetConfiguration = linkTarget.Configuration ]
+        [#local linkTargetResources = linkTarget.State.Resources ]
+        [#local linkTargetAttributes = linkTarget.State.Attributes ]
+
+        [#switch linkTargetCore.Type]
+            [#case DATASET_COMPONENT_TYPE]
+                [#if linkTargetConfiguration.Solution.Engine == "dds" ]
+                    [#local ddsManualSnapshot = linkTargetAttributes["SNAPSHOT_NAME"] ]
+                [/#if]
+                [#break]
```

Comment:
Don't think we need to include DATASETS here.

Reply:
Removed. I can't remember why I thought we did, so I've left the structure but removed DATASETS